packages/core/src/extensions/SuggestionMenu/getDefaultEmojiPickerItems.ts

```diff
@@ -0,0 +1,37 @@
+import data, { Emoji, EmojiMartData } from "@emoji-mart/data";
+import { init, SearchIndex } from "emoji-mart";
+
+import { BlockSchema, InlineContentSchema, StyleSchema } from "../../schema";
+import { BlockNoteEditor } from "../../editor/BlockNoteEditor";
+import { checkDefaultInlineContentTypeInSchema } from "../../blocks/defaultBlockTypeGuards";
+import { DefaultGridSuggestionItem } from "./DefaultGridSuggestionItem";
+
+const emojiMartData = data as EmojiMartData;
+init({ emojiMartData });
+
+export async function getDefaultEmojiPickerItems<
+  BSchema extends BlockSchema,
+  I extends InlineContentSchema,
+  S extends StyleSchema
+>(
+  editor: BlockNoteEditor<BSchema, I, S>,
+  query: string
+): Promise<DefaultGridSuggestionItem[]> {
+  if (!checkDefaultInlineContentTypeInSchema("text", editor)) {
+    return [];
+  }
+
+  const emojisToShow =
+    query === ""
+      ? Object.values(emojiMartData.emojis)
+      : ((await SearchIndex.search(query)) as Emoji[]);
+
+  return emojisToShow.map((emoji: Emoji) => ({
+    id: emoji.skins[0].native,
+    onItemClick: () => {
+      if (checkDefaultInlineContentTypeInSchema("text", editor)) {
```

Comment:
do we need this check?

Reply:
In theory yes as we can't insert an emoji unless `text` is in the inline content schema. In practice, I don't think pretty much anyone is likely to use the editor without the `text` IC. But eh imo it's kind of the same as with the paragraph block in the slash menu items - we also check if that's in the schema (despite 99% of people using the paragraph block) and there's not really any issues there.